zerver/lib/test_runner.py

```diff
@@ -567,7 +567,7 @@ def get_test_names(suite: unittest.TestSuite) -> List[str]:
         # first element is the type of TestSuite and the second element is a
         # list of test names in that test suite. See serialize_suite() for the
         # implementation details.
-        return [name for subsuite in suite.subsuites for name in subsuite[1]]
+        return [name for subsuite in suite.subsuites for name in subsuite[1]]  # type: ignore # ->warn-unreachable
```

Comment:
Is this because mypy isn’t aware that `django_runner.ParallelTestSuite` is a subclass of `unittest.TestSuite`?

Maybe we want to type `suite` as `Union[TestSuite, django_runner.ParallelTestSuite]`, and then we can also get rid of the unchecked `cast` below?

Reply:
Mypy should be able to see the inheritance there, right?  Which lead me to think that this was because we are checking whether a parameter typed as `unittest.TestSuite` is an instance of `ParallelTestSuite`?   In any case, what you're suggesting sounds like a good way to go.

Reply:
> Mypy should be able to see the inheritance there, right?

We don’t have type stubs for `django`.

> Which lead me to think that this was because we are checking whether a parameter typed as `unittest.TestSuite` is an instance of `ParallelTestSuite`?

That would be fine, if Mypy understood that `ParallelTestSuite` is a subclass of `unittest.TestSuite`.

Reply:
Oh, right. Woops.. 🙊